measuring-client/src/main/java/de/cyface/app/ui/button/DataCapturingButton.java

```diff
@@ -901,50 +912,50 @@ public void onFixLost() {
     @Override
     public void onNewGeoLocationAcquired(GeoLocation geoLocation) {
         Log.d(TAG, "onNewGeoLocationAcquired");
-        final Measurement measurement;
+
         try {
-            measurement = persistenceLayer.loadCurrentlyCapturedMeasurement();
-        } catch (final NoSuchMeasurementException e) {
-            // GeoLocations may also arrive shortly after a measurement was stopped. Thus, this may not crash.
-            // This happened on the Emulator with emulated live locations.
-            Log.w(TAG, "onNewGeoLocationAcquired: No currently captured measurement found, doing nothing.");
+            final Measurement measurement = persistenceLayer.loadCurrentlyCapturedMeasurement();
+
+            final int distanceMeter = (int)Math.round(measurement.getDistance());
+            final double distanceKm = distanceMeter == 0 ? 0.0 : distanceMeter / 1000.0;
+            final String distanceText = distanceKm + " km";
+            Log.d(TAG, "Distance update: " + distanceText);
+            distanceTextView.setText(distanceText);
+
+            final long durationMillis = persistenceLayer.loadDuration(measurement.getIdentifier());
+            final long durationSeconds = durationMillis / 1000;
+            final long durationMinutes = durationSeconds / 60;
+            final long durationHours = durationMinutes / 60;
+            final String hoursText = durationHours > 0 ? durationHours + "h " : "";
+            final String minutesText = durationMinutes > 0 ? durationMinutes % 60 + "m " : "";
+            final String secondsText = durationSeconds % 60 + "s";
+            final String durationText = hoursText + minutesText + secondsText;
+            Log.d(TAG, "Duration update: " + durationMillis);
+            durationTextView.setText(durationText);
```

Comment:
The `Duration` class is only available starting at Android 8.1, but we're still targeting 5.0.

Thus, I wrote this simple code here to format the Duration. Output is:

- 1s
- 2s
- ...
- 59s
- 1m 0s
- ...
- 1h 0m 0s
- ...

Reply:
But this is anyway just for the Test-App (App V2), i.e. only to test SDK 6

Reply:
Yeah I would like to switch to MVVM via LiveData, Room, etc. on Android but that would be a larger refactoring.
Maybe when we design the next UI (planned) we should try to switch to LiveData and Room there, but I don't want to change patterns here in the SDK6-Test-App which is not used anywhere